onnxruntime/core/providers/webnn/builders/impl/max_min_op_builder.cc

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Copyright (c) Intel Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/common.h"
+#include "core/providers/shared/utils/utils.h"
+#include "core/providers/webnn/builders/helper.h"
+#include "core/providers/webnn/builders/model_builder.h"
+#include "core/providers/webnn/builders/op_builder_factory.h"
+
+#include "base_op_builder.h"
+
+namespace onnxruntime {
+namespace webnn {
+
+class MaxMinOpBuilder : public BaseOpBuilder {
+  // Add operator related.
+ private:
+  Status AddToModelBuilderImpl(ModelBuilder& model_builder, const Node& node,
+                               const logging::Logger& logger) const override ORT_MUST_USE_RESULT;
+
+  // Operator support related.
+  bool IsOpSupportedImpl(const InitializedTensorSet& initializers, const Node& node,
+                         WebnnDeviceType /* device_type */, const logging::Logger& logger) const override;
+};
+
+// Add operator related.
+
+Status MaxMinOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
+                                                 const Node& node,
+                                                 const logging::Logger& logger) const {
+  const auto& input_defs = node.InputDefs();
+  const auto& op_type = node.OpType();
+
+  emscripten::val input0 = model_builder.GetOperand(input_defs[0]->Name());
+  std::vector<int64_t> input_shape;
+  ORT_RETURN_IF_NOT(GetShape(*input_defs[0], input_shape, logger), "Cannot get shape");
+
+  const size_t input_count = input_defs.size();
+  ORT_RETURN_IF(input_count < 1, op_type, "has no inputs");
+
+  emscripten::val output = emscripten::val::object();
+
+  if (input_count == 1) {
+    output = input0;
```

Comment:
@fdwr, this would fail  with some unit test  with a single op model, e.g. https://github.com/onnx/onnx/tree/main/onnx/backend/test/data/node/test_min_one_input

Because in this situation there's no op in the graph of WebNN (only one input and one output). 

Since WebNN has no identity op at present, shall we use some op to workaround it? e.g. concat a single input, split the input to one output, etc. These operations in the underlying implementation use the Identity op from backends.

Reply:
Does ORT optimize such a Min case into an identity or empty graph before reaching the WebNN EP?

Note I proposed adding `identity` for WebNN v2, and my fork supports `identity`. For master chrome though, maybe use a Concat with a single input or a Reshape to the same shape?

Reply:
Yeah, I will use Concat as workaround until we have `identity` available. :)